### PR TITLE
Support go-restful framework with v3 version

### PR DESCRIPTION
**Pull request by mrproliu**
1. Support go-restful framework with v3 version
2. Support overwide operation names avoid to create multiple entry span
3. Support overwrite span tag

### CODE REVIEW

**Comment by wu-sheng** [comment]
> Support overwide operation names to create multiple entry span

`to create multiple` or `avoid to create multiple`?

**Comment by mrproliu** [comment]
> > Support overwide operation names to create multiple entry span
> 
> `to create multiple` or `avoid to create multiple`?

is `avoid to create multiple`, I have updated the description.

File: plugins/core/span_tracing.go [hunk 1]
```diff
@@ -28,6 +28,11 @@ import (
 )
 
 func NewSegmentSpan(defaultSpan *DefaultSpan, parentSpan SegmentSpan) (s SegmentSpan, err error) {
+	// if current and parent span are both entry span, then use parent span as segment span
+	if defaultSpan.SpanType == SpanTypeEntry && parentSpan != nil && parentSpan.IsEntry() {
```
Comment (wu-sheng) [inline plugins/core/span_tracing.go:32]: Should we check parent span first before the span creation? It seems to cause a little extra payload?

Comment (mrproliu) [reply plugins/core/span_tracing.go:32]: I have advanced the parent check, please re-check

File: plugins/core/tracing.go [hunk 2]
```diff
@@ -47,6 +47,11 @@ func (t *Tracer) CreateEntrySpan(operationName string, extractor interface{}, op
 	defer func() {
 		saveSpanToActiveIfNotError(ctx, s, err)
 	}()
+	// if parent span is entry span, then use parent span as result
+	if tracingSpan != nil && tracingSpan.IsEntry() {
+		tracingSpan.SetOperationName(operationName)
+		return tracingSpan, nil
+	}
```
Comment (wu-sheng) [inline plugins/core/tracing.go:54]: Do we have exit span overlap mechanism? In Java agent, the entry span is always using the newest created, but exit span usually uses the first time set.

**Review by wu-sheng — APPROVED** [review]
LGTM

**Review by wu-sheng — APPROVED** [review]
LGTM

File: plugins/core/tracing.go (hunk 2 above)
Comment (mrproliu) [reply plugins/core/tracing.go:54]: For now, I haven't added the exit span overlap mechanism, I can add it in other PR.

**Pull request merged**